src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ArrayList.cs

```diff
@@ -0,0 +1,103 @@
+﻿using global::System;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	/// <summary>
+	/// Simplified implementation of List<typeparamref name="T"/> that supports direct access to the internal array.
+	/// </summary>
+	/// <remarks>
+	/// This is needed in the ColorPicker to avoid an extra copy when converting a list of bytes into a WritableBitmap.
+	/// The performance gains justify creation of this class and a slight deviation from WinUI.
+	/// This class is not considered safe to use outside of the ColorPicker so it is not in a generic location.
+	/// </remarks>
+	/// <typeparam name="T">The type of elements in the list.</typeparam>
+	internal class ArrayList<T>
+	{
+		private int _Capacity;
+		private T[] _Array;
+
+		public ArrayList()
+		{
+			_Capacity = 0;
+			_Array = global::System.Array.Empty<T>();
+		}
+
+		public ArrayList(int capacity)
+		{
+			if (capacity < 0)
+			{
+				throw new ArgumentOutOfRangeException("Capacity cannot be less than zero.");
+			}
+			else if (capacity == 0)
+			{
+				_Array = global::System.Array.Empty<T>();
+			}
+			else
+			{
+				_Array = new T[capacity];
+			}
+
+			_Capacity = capacity;
+		}
+
+		internal T[] Array
+		{
+			get => _Array;
+		}
+
+		public int Capacity
+		{
+			get => _Capacity;
+			set
+			{
+				if (value < Count)
+				{
+					throw new IndexOutOfRangeException("Cannot resize the list smaller than its current length.");
```

Comment:
Codacy found an issue: ['System.IndexOutOfRangeException' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)